static/js/store/edges.js

```diff
@@ -23,14 +24,36 @@ var app = app || {};
     Edge.prototype.constructor = Edge;
 
     Edge.prototype.geometry = function() {
+
         this.dirtyPicking = true;
         var from = app.NodeStore.getVisibleParent(this.idFrom);
         var to = app.NodeStore.getVisibleParent(this.idTo);
+        
+        // if either of the routes don't exist any more, simply return
+        if (!(this.routeIdFrom in from.routeGeometry)){
+          this.visible = false
+          return 
+        }
+        if (!(this.routeIdTo in to.routeGeometry)){
+          this.visible = false
+          return 
+        }
 
         // TODO: organize -- this is so terribly ugly
         var fromV = app.NodeStore.getNode(app.NodeStore.getRoot()).data.children.indexOf(from.visibleParent) + 1;
         var toV = app.NodeStore.getNode(app.NodeStore.getRoot()).data.children.indexOf(to.visibleParent) + 1;
-        this.visible = !!fromV && !!toV;
+        //
+        var fromH = false
+        if (from instanceof app.Group){
+          fromH = from.data.hiddenRoutes.indexOf(this.routeIdFrom) != -1
+        } 
+        //
+        var toH = false
+        if (to instanceof app.Group){
+          toH = to.data.hiddenRoutes.indexOf(this.routeIdTo) != -1
+        }
+
+        this.visible = (!!fromV && !!toV) && (from.data.id != to.data.id) && (!fromH && ! toH)
```

Comment:
but `from.data.id` and `to.data.id` hold block ids, not the enclosing group id, right?